Libraries/LibURL/Host.cpp

```diff
@@ -193,8 +193,12 @@ Optional<String> Host::public_suffix() const
     // 3. Let publicSuffix be the public suffix determined by running the Public Suffix List algorithm with host as domain. [PSL]
     auto public_suffix = get_public_suffix(host_string.bytes_as_string_view());
 
-    // NOTE: get_public_suffix() returns Optional, but this algorithm assumes a value. Is that OK?
-    VERIFY(public_suffix.has_value());
+    // AD-HOC: This algorithm assumes a public suffix is returned, but our implementation can return null.
+    //         In this case, return null instead of crashing.
+    // FIXME: What's the correct behaviour here?
```

Comment:
I think the [Public Suffix List algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm) is supposed to return `*` if no rules are matched. I'm not sure whether that would work with our current implementation though.

Tested using:
```sh
export DOMAIN=localhost
dig +short PTR $DOMAIN.query.publicsuffix.zone" | tail -n1
```

Reply:
Thanks. I've modified this and the other user of `get_public_suffix()` here to use "*" as the default instead. That seems more correct.

Probably `get_public_suffix()` should itself be modified, but I'd rather leave that for now.